devicesetup/src/main/java/io/particle/android/sdk/devicesetup/ApConnector.java

```diff
@@ -147,6 +146,14 @@ public SSID connectToAP(final WifiConfiguration config, Client client) {
             setupRunnables.add(() -> {
                 log.d("Adding network " + configSSID);
                 networkID.set(wifiFacade.addNetwork(config));
+
+                if (networkID.get() == -1) {
```

Comment:
Sometimes we used to get `-1` on `addNetwork` in Nougat, not really succeeding when it fails to add network.